thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/UserDashboardResource.java

```diff
@@ -0,0 +1,217 @@
+import com.linkedin.thirdeye.datalayer.bao.MergedAnomalyResultManager;
+import com.linkedin.thirdeye.datalayer.dto.AlertConfigDTO;
+import com.linkedin.thirdeye.datalayer.dto.AnomalyFunctionDTO;
+import com.linkedin.thirdeye.datalayer.dto.MergedAnomalyResultDTO;
+import com.linkedin.thirdeye.datalayer.util.Predicate;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+
+
+/**
+ * Endpoints for user-customized dashboards (currently alerts only)
+ */
+@Path(value = "/userdashboard")
+@Produces(MediaType.APPLICATION_JSON)
+public class UserDashboardResource {
+  private final MergedAnomalyResultManager anomalyDAO;
+  private final AnomalyFunctionManager functionDAO;
+  private final AlertConfigManager alertDAO;
+
+  public UserDashboardResource(MergedAnomalyResultManager anomalyDAO, AnomalyFunctionManager functionDAO,
+      AlertConfigManager alertDAO) {
+    this.anomalyDAO = anomalyDAO;
+    this.functionDAO = functionDAO;
+    this.alertDAO = alertDAO;
+  }
+
+  /**
+   * Returns a list of AnomalySummary for a set of query parameters. Anomalies are
+   * sorted by start time (descending).
+   *
+   * <br/><b>Example:</b>
+   * <pre>
+   *   [ {
+   *     "id" : 12345,
+   *     "start" : 1517000000000,
+   *     "end" : 1517100000000,
+   *     "dimensions" : { "country": "us" },
+   *     "severity" : 0.517,
+   *     "current" : 1213.0,
+   *     "baseline" : 550.0,
+   *     "feedback" : "NO_FEEDBACK",
+   *     "metric" : "page_views",
+   *     "functionName" : "page_views_monitoring"
+   *     },
+   *     ...
+   *   ]
+   * </pre>
+   *
+   * @see AnomalySummary
+   *
+   * @param start window start time
+   * @param end window end time (optional)
+   * @param owner anomaly function owner only (optional)
+   * @param application anomaly function for application alert groups only (optional)
+   *
+   * @return List of AnomalySummary
+   */
+  @GET
+  @Path("/anomalies")
+  public List<AnomalySummary> queryAnomalies(
+      @QueryParam("start") Long start,
+      @QueryParam("end") Long end,
+      @QueryParam("owner") String owner,
+      @QueryParam("application") String application,
+      @QueryParam("limit") Integer limit) {
+
+    //
+    // query params
+    //
+
+    // safety condition
+    Preconditions.checkNotNull(start);
+
+    List<Predicate> predicates = new ArrayList<>();
+
+    // TODO support index select on user-reported anomalies
+//    predicates.add(Predicate.OR(
+//        Predicate.EQ("notified", true),
+//        Predicate.EQ("anomalyResultSource", AnomalyResultSource.USER_LABELED_ANOMALY)));
+
+
+    // application (indirect)
```

Comment:
For the part of joining application with the anomaly function, can you move this part to an util or a static method, as we will use this method for many other endpoints.

Reply:
This is a simple code fragment. Moving this to a separate method creates way more clutter than it saves

Reply:
I can understand, but this code segment is shared by multiple methods, for the sake of code re-usability, or easier for us to improve the performance in the future. In my opinion, I think we should better utilize the code segment.

Reply:
Where would you put this method, and what would be it's signature?

Reply:
I'm thinking of having this method in ApplicationManager, where we can fetch the list of anomaly function ids from the DAO.

Reply:
If this were to go into a DAO, then it would be the AlertConfig DAO. That wouldn't be the place where anyone would look for it though - i.e. this is a classic case of cross-entity business logic.

Reply:
In this sense how do we come up with a generic way to fetch anomalies under an application. Noting this can be executed drastically in the dashboard that we are going to build. How can we make sure avoid code duplications and improve execution efficiency?

Reply:
Please move this code segment to an util method. I know this code segment may be simple. I still hope the implementation can be simplified.

Reply:
I gladly invite your contribution to this PR. Please refactor the code as you see fit.

Reply:
Please change the code followed by https://github.com/linkedin/pinot/pull/2650.